ws.go

```diff
@@ -52,75 +75,137 @@ func doAsyncTimeout(f func() error, tmFunc func(error), timeout time.Duration) e
 	}
 }
 
-func sendStateAsync(dataCh chan<- ExchangeState, st ExchangeState) {
-	select {
-	case dataCh <- st:
-	default:
-	}
-}
-
 func subForMarket(client *signalr.Client, market string) (json.RawMessage, error) {
 	_, err := client.CallHub(WS_HUB, "SubscribeToExchangeDeltas", market)
 	if err != nil {
 		return json.RawMessage{}, err
 	}
+
 	return client.CallHub(WS_HUB, "QueryExchangeState", market)
 }
 
-func parseStates(messages []json.RawMessage, dataCh chan<- ExchangeState, market string) {
+func parseStates(messages []json.RawMessage, dataCh chan<- ExchangeState, markets map[string]bool) error {
 	for _, msg := range messages {
 		var st ExchangeState
 		if err := json.Unmarshal(msg, &st); err != nil {
-			continue
+			return err
+		}
+
+		if _, ok := markets[st.MarketName]; ok {
+			dataCh <- st
 		}
-		if st.MarketName != market {
-			continue
+	}
+
+	return nil
+}
+
+func parseDeltas(messages []json.RawMessage, dataCh chan<- SummaryState, markets map[string]bool) error {
+	for _, msg := range messages {
+		var d ExchangeDelta
+		if err := json.Unmarshal(msg, &d); err != nil {
+			return err
 		}
-		sendStateAsync(dataCh, st)
+
+		for _, v := range d.Deltas {
+			if _, ok := markets[v.MarketName]; ok {
+				dataCh <- v
+			}
+		}
+	}
+
+	return nil
+}
+
+func (b *Bittrex) SubscribeSummaryUpdate(markets []string, dataCh chan<- SummaryState, stop <-chan bool) error {
+	const timeout = 5 * time.Second
+	m := make(map[string]bool, len(markets))
+	for _, market := range markets {
+		m[market] = false
+	}
+
+	client := signalr.NewWebsocketClient()
+	client.OnClientMethod = func(hub string, method string, messages []json.RawMessage) {
+		if hub != WS_HUB || method != "updateSummaryState" {
+			return
+		}
+
+		parseDeltas(messages, dataCh, m)
+	}
+	defer client.Close()
```

Comment:
Disable your internet connection and run the tests. You'll get
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x655d22]
goroutine 7 [running]:
github.com/gorilla/websocket.(*Conn).Close(0x0, 0xc420001828, 0xc42012ee58)
	/home/avd/dev/godev/src/github.com/gorilla/websocket/conn.go:304 +0x22
github.com/thebotguys/signalr.(*Client).Close(0xc42011e0a0)
	/home/avd/dev/godev/src/github.com/thebotguys/signalr/client.go:268 +0x2fexit status 2
FAIL	github.com/KLIM8D/go-bittrex	0.005s

github.com/KLIM8D/go-bittrex.(*Bittrex).SubscribeSummaryUpdate(0xc42000e050, 0x87cba0, 0x2, 0x2, 0xc420062240, 0xc42006c360, 0x84e3a0, 0xc4201643f0)
	/home/avd/dev/godev/src/github.com/KLIM8D/go-bittrex/ws.go:142 +0x354
```
signalr.NewWebsocketClient() returns uninitialized socket, and if Connect() failes, Close() will try to close nil conn.
I do `defer client.Close()` only after a call to Connect(), `doAsyncTimeout`, on the other side, was designed to close the conn after timeout had elapsed, to avoid conn leaks.